emperor/support_files/js/controller.js

```diff
@@ -1,37 +1,177 @@
-EmperorController = function(width, height, dm, div_id){
-  this.width = width;
-  this.height = height;
+/**
+ *
+ * @author Jamie Morton, Jose Navas Molina, Andrew Hodges & Yoshiki
+ *         Vazquez-Baeza
+ * @copyright Copyright 2013--, The Emperor Project
+ * @credits Jamie Morton, Jose Navas Molina, Andrew Hodges & Yoshiki
+ *          Vazquez-Baeza
+ * @license BSD
+ * @version 0.9.51-dev
+ * @maintainer Jose Antonio Navas Molina
+ * @email josenavas@gmail.com
+ * @status Development
+ *
+ */
+
+EmperorController = function(dm, divId){
+  this.$divId = $('#' + divId);
+  this.width = this.$divId.width();
+  this.height = this.$divId.height();
+
+  this.dm = dm;
+  this.sceneViews = [];
+
+  // main divs where the content of the plots will be located
+  this.$plotSpaceId = $("<div id='main-wrapper' class='emperor-plot-wrapper'></div>");
+  this.$plotMenu = $("<div id='emperor-menu'></div>");
+
+  this.$divId.append(this.$plotSpaceId);
+  this.$divId.append(this.$plotMenu);
 
   // set up the renderer
   this.rendererBackgroundColor = new THREE.Color();
-  this.rendererBackgroundColor.setHex("0x000000");
+  this.rendererBackgroundColor.setHex('0x000000');
 
-  this.renderer = new THREE.WebGLRenderer( {antialias: true});
-  this.renderer.setSize(width, height);
+  this.renderer = new THREE.WebGLRenderer({antialias: true});
+  this.renderer.setSize(this.width, this.height);
   this.renderer.setClearColor(this.rendererBackgroundColor);
   this.renderer.autoClear = false;
   this.renderer.sortObjects = true;
+  this.$plotSpaceId.append(this.renderer.domElement);
 
-  this.sceneViews = [
-    new ScenePlotView3D(this.renderer, [new DecompositionView(dm)], div_id, 0, 0, 0.5 * this.width, 0.5 * this.height),
-    new ScenePlotView3D(this.renderer, [new DecompositionView(dm)], div_id, 0.5 * this.width, 0, 0.5 * this.width, 0.5 * this.height),
-    new ScenePlotView3D(this.renderer, [new DecompositionView(dm)], div_id, 0, 0.5 * this.height, 0.5 * this.width, 0.5 * this.height),
-    new ScenePlotView3D(this.renderer, [new DecompositionView(dm)], div_id, 0.5 * this.width, 0.5 * this.height, 0.5 * this.width, 0.5 * this.height)];
+  // default decomposition view uses the full window
+  this.addView();
+
+  this.buildUI();
+};
+
+EmperorController.prototype.addView = function() {
+  if (this.sceneViews.length > 4) {
+    throw Error('Cannot add another scene plot view');
```

Comment:
Would anyone be opposed to adding in an immutable global variable `maxScenePlotViews=4`?  

Just in case we may want to add more scene plot views in the future, it'll be easier to tune all of these constants in a single place, rather than digging through all of the source files.